zrml/global-disputes/src/global_disputes_pallet_api.rs

```diff
@@ -17,46 +17,27 @@
 
 extern crate alloc;
 
-use alloc::vec::Vec;
-use sp_runtime::DispatchResult;
+use crate::types::InitialItem;
+use sp_runtime::DispatchError;
 use zeitgeist_primitives::types::OutcomeReport;
 
 /// The trait to initiate and resolve the global disputes.
-pub trait GlobalDisputesPalletApi<MarketId, AccountId, Balance> {
-    /// Push a voting outcome for one global dispute.
-    ///
-    /// # Arguments
-    /// - `market_id` - The id of the market.
-    /// - `outcome` - The voting outcome to push.
-    /// - `owner` - The owner of the outcome.
-    /// - `initial_vote_balance` - The initial vote amount for the specified outcome.
-    ///
-    /// # Returns
-    ///
-    /// Returns the dispute mechanism's report if available, otherwise `None`. If `None` is
-    /// returned, this means that the dispute could not be resolved.
-    fn push_voting_outcome(
-        market_id: &MarketId,
-        outcome: OutcomeReport,
-        owner: &AccountId,
-        initial_vote_balance: Balance,
-    ) -> DispatchResult;
+pub trait GlobalDisputesPalletApi<MarketId, AccountId, Balance, BlockNumber> {
+    /// Return the `AddOutcomePeriod` parameter.
+    fn get_add_outcome_period() -> BlockNumber;
+
+    /// Return the `GdVotingPeriod` parameter.
+    fn get_vote_period() -> BlockNumber;
 
-    /// Get the information about a voting outcome for a global dispute.
+    /// Start a global dispute.
     ///
     /// # Arguments
     /// - `market_id` - The id of the market.
-    /// - `outcome` - The voting outcome to get.
-    ///
-    /// # Returns
-    ///
-    /// Returns the information stored for a particular outcome.
-    /// - outcome_sum - The current sum of all locks on this outcome.
-    /// - owners - The vector of owners of the outcome.
-    fn get_voting_outcome_info(
+    /// - `initial_items` - The initial items (outcome, owner, amount) to add to the global dispute.
+    fn start_global_dispute(
```

Comment:
This is "write everything twice" documentation. I know that `start_global_dispute` starts a global dispute, because that's what the name of the function is. I know that `market_id` is the id of the market because that's what the parameter name is. You might as well not mention these things.

On the other hand, here are a couple of things I don't understand: What are _initial items_? How many do I have to specify? Can I just not provide any? What if there's a duplicate? What if only some fields coincide (for example, what if two initial items have matching `outcome` and `owner` fields but different amounts? Etc. These are the things that actually need to be documented.

Reply:
56cd5328f631a689da5de504b6228aa3e70387e7

Reply:
> On the other hand it seems inconsistent to just document a few parameters instead of all of these.

Been there. Not a fan of this situation either. In private projects, I've given up on documenting parameters completely and just providing details which aren't obvious. For example: What happens if there's a duplicate? What happens if there's only one unique outcome, etc.

Reply:
If you look at Rust's documentation, you'll almost always find that style of documentation:

> [Short imperative sentence describing the function.]
>
> [One ore more paragraphs describing the function's details.]

If a parameter's function is unclear or complex, you'll find an explanation in one of the paragraphs.

Reply:
You seem to rename the parameter into `initial_item` here. Seems like a typo.